### PR TITLE
fontconfig: Add CoreFoundation to buildInputs

### PR DESCRIPTION
**Pull request by hannesweisbach**
###### Motivation for this change

fontconfig depends on CoreFoundation, but did not have CoreFoundation in its buildInputs. This built fine, but resulted in a dylib with a non-absolute rpath as reference:

```
otool -L /nix/store/zy90l8cw2qj316xdadhmpf9i3wyisn4g-fontconfig-2.13.92-lib/lib/libfontconfig.dylib
/nix/store/zy90l8cw2qj316xdadhmpf9i3wyisn4g-fontconfig-2.13.92-lib/lib/libfontconfig.dylib:
	/nix/store/zy90l8cw2qj316xdadhmpf9i3wyisn4g-fontconfig-2.13.92-lib/lib/libfontconfig.1.dylib (compatibility version 14.0.0, current version 14.0.0)
	/nix/store/j7m6j79aw01gs1kqiv7n8gdgaj3kdjdn-freetype-2.10.2/lib/libfreetype.6.dylib (compatibility version 24.0.0, current version 24.2.0)
	/nix/store/ik820r3rrfr7x7k1iyc6w9xj6mhinynm-bzip2-1.0.6.0.1/lib/libbz2.1.dylib (compatibility version 2.0.0, current version 2.6.0)
	/nix/store/xr4cvl6blqrs6vwklnigpb43h4jqz04k-libpng-apng-1.6.37/lib/libpng16.16.dylib (compatibility version 54.0.0, current version 54.0.0)
	/nix/store/jmbhjg7yzgpxbhvj87n3svkqsvdvmbi7-zlib-1.2.11/lib/libz.dylib (compatibility version 1.0.0, current version 1.2.11)
	/nix/store/bqhgv3ppas6gbdpspb9yqdfshspb9rkd-expat-2.2.8/lib/libexpat.1.dylib (compatibility version 8.0.0, current version 8.10.0)
	/nix/store/bpfk6g2y6a7jilf55p08pbdajjq6ha2p-gettext-0.20.1/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.6.0)
	@rpath/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)
	/nix/store/bwny9pyjhghmv8jq5836ms422iwiqi5l-Libsystem-osx-10.12.6/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

With CoreFoundation as buildInput the dylib has only absolute paths to references:

```
otool -L /nix/store/n1c1ghp21c0n7c5a8vc4fqjmaysyvgfr-fontconfig-2.13.92-lib/lib/libfontconfig.dylib
/nix/store/n1c1ghp21c0n7c5a8vc4fqjmaysyvgfr-fontconfig-2.13.92-lib/lib/libfontconfig.dylib:
	/nix/store/n1c1ghp21c0n7c5a8vc4fqjmaysyvgfr-fontconfig-2.13.92-lib/lib/libfontconfig.1.dylib (compatibility version 14.0.0, current version 14.0.0)
	/nix/store/slm1f903cz47r2hi8v543mj3fcakvwrc-freetype-2.10.2/lib/libfreetype.6.dylib (compatibility version 24.0.0, current version 24.2.0)
	/nix/store/bafwypxc11wp51phssmh9h6vqmwzmnxf-bzip2-1.0.6.0.1/lib/libbz2.1.dylib (compatibility version 2.0.0, current version 2.6.0)
	/nix/store/0hr8d4x5w1046a6mcwgfnyj85f7ylf8g-libpng-apng-1.6.37/lib/libpng16.16.dylib (compatibility version 54.0.0, current version 54.0.0)
	/nix/store/ilrm722frr4fb3096wjc6wbspdf7pgyr-zlib-1.2.11/lib/libz.dylib (compatibility version 1.0.0, current version 1.2.11)
	/nix/store/cjflr1cvpkq74ln7j2qndri3a5513797-expat-2.2.8/lib/libexpat.1.dylib (compatibility version 8.0.0, current version 8.10.0)
	/nix/store/vrxrz8fqi2cmv0z7kxfza2nycbbbkh1k-gettext-0.21/lib/libintl.8.dylib (compatibility version 11.0.0, current version 11.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1677.104.0)
	/nix/store/hw4x6gakgd43f13zys92ng0hj0acisls-Libsystem-osx-10.12.6/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

Fixes (at least in part) #98203
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/fontconfig/default.nix [hunk 1]
```diff
@@ -69,7 +70,7 @@ stdenv.mkDerivation rec {
     libxslt
     pkg-config
     autoreconfHook
-  ];
+  ] ++ stdenv.lib.optional stdenv.isDarwin CoreFoundation;
```
Comment (jtojnar) [inline pkgs/development/libraries/fontconfig/default.nix:73]: Did you mean to add it to `buildInputs`?

Comment (hannesweisbach) [reply pkgs/development/libraries/fontconfig/default.nix]: Yes, thanks. Still on my first coffee today ;)

**Comment by prusnak** [comment]
Since this will trigger a massive rebuild on Darwin, this should be targeted on `staging` branch, right @jtojnar ?

**Review by veprbl — APPROVED** [review]

File: pkgs/development/libraries/fontconfig/default.nix [hunk 2]
```diff
@@ -73,7 +74,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     expat
-  ];
+  ] ++ stdenv.lib.optional stdenv.isDarwin CoreFoundation;
```
Comment (prusnak) [inline pkgs/development/libraries/fontconfig/default.nix:77]: We don't use `stdenv.lib` anymore. Please use the following and at add `lib` at the top of the file

```suggestion
  ] ++ lib.optional stdenv.isDarwin CoreFoundation;
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/fontconfig/default.nix [hunk 3]
```diff
@@ -74,7 +74,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     expat
-  ] ++ stdenv.lib.optional stdenv.isDarwin CoreFoundation;
+  ] ++ lib.optional stdenv.isDarwin CoreFoundation;
```
Comment (hannesweisbach) [inline pkgs/development/libraries/fontconfig/default.nix:77]: Thanks for taking care of this! I might be wrong, but aren't you missing "lib" as parameter at the top of the file (first pair of curlies?, Sorry, I don't know how it is called …)

Like so:
```
{ stdenv
, lib
…
}
```

Comment (prusnak) [reply pkgs/development/libraries/fontconfig/default.nix:77]: Everything is allright now in the `staging` branch (see https://github.com/NixOS/nixpkgs/blob/staging/pkgs/development/libraries/fontconfig/default.nix). Probably there was a merge of different commit or something like that.

Comment (hannesweisbach) [reply pkgs/development/libraries/fontconfig/default.nix:77]: Oh, ok, that make sense, since the other stdenv.lib got probably fixed there.

**Pull request merged**